src/pages/main/components/sidebar/datasets-list/components/dataset-name.tsx

```diff
@@ -28,10 +28,14 @@ export const DatasetName: FC<IDsNameProps> = ({
 
   const name = isXL && /^xl_.*/i.test(dsName) ? dsName.substring(3) : dsName
 
-  const { x = 0, width = 0 } =
-    datasetRef?.current?.getBoundingClientRect() || {}
+  const [hasTooltip, setHasTooltip] = useState<boolean>(false)
 
-  const hasTooltip = width + x > 230
+  useEffect(() => {
```

Comment:
I think it's better to decide whether a tooltip is needed is to check that text doesn't fit into container.
container.clientWidth would be less than container.scrollWidth if text doesn't fit into container with overflow: hidden

Reply:
Please, try to use it